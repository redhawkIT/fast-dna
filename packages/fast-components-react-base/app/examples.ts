import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };
