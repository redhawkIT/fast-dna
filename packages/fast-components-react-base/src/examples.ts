import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: Array<IExampleData<T>>;
}
