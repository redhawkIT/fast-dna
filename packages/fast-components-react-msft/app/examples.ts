import ButtonExample from "../src/button/examples.data";
export {ButtonExample};

import DividerExample from "../src/divider/examples.data";
export {DividerExample};

import ToggleExamples from "../src/toggle/examples.data";
export {ToggleExamples};
