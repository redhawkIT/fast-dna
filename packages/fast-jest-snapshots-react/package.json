{
  "name": "@microsoft/fast-jest-snapshots-react",
  "description": "A small utility for automating Jest snapshot creation of React components",
  "version": "1.6.0",
  "author": {
    "name": "Microsoft",
    "email": "fwq@microsoft.com"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Microsoft/fast-dna.git"
  },
  "private": true,
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "prepare": "tsc -p ./tsconfig.json",
    "test": "npm run tslint",
    "start": "webpack-dev-server",
    "tslint": "tslint -c ./tslint.json '**/*.ts{,x}'",
    "tslint:fix": "tslint -c ./tslint.json --fix '**/*.ts{,x}'"
  },
  "devDependencies": {
    "@microsoft/fast-tslint-rules": "^1.6.0",
    "@types/jest": "^22.2.2",
    "jest": "^22.4.3",
    "react": "^16.3.0",
    "ts-jest": "^22.4.2",
    "tslint": "^5.9.1",
    "typescript": "^2.8.1"
  },
  "dependencies": {
    "@types/react": "^16.3.0",
    "react-test-renderer": "^16.3.0"
  },
  "peerDependencies": {
    "jest": "^22.0.0",
    "react": "^16.3.0"
  }
}
