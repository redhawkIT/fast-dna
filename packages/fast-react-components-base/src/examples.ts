import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: IExampleData<T>[];
}