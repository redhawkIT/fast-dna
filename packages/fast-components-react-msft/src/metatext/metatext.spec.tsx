import * as React from "react";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import examples from "./examples.data";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import MSFTMetatext, {
    IMetatextHandledProps,
    IMetatextManagedClasses,
    IMetatextUnhandledProps,
    MetatextProps,
    MetatextTag
} from "./metatext";
import { Metatext } from "./index";

/*
 * Configure Enzyme
 */
configure({adapter: new Adapter()});

describe("metatext snapshots", (): void => {
    generateSnapshots(examples);
});

describe("metatext", (): void => {
    test("should have a displayName that matches the component name", () => {
        expect((MSFTMetatext as any).name).toBe(MSFTMetatext.displayName);
    });

    test("should return an object that includes all valid props which are not enumerated as handledProps", () => {
        const handledProps: IMetatextHandledProps = {
            tag: MetatextTag.p
        };

        const unhandledProps: IMetatextUnhandledProps = {
            "aria-hidden": true
        };

        const props: IMetatextHandledProps & IMetatextUnhandledProps = {...handledProps, ...unhandledProps};

        const rendered: any = shallow(
            <Metatext {...props} />
        );

        const paragraph: any = rendered.first().shallow();

        expect(paragraph.prop("aria-hidden")).toEqual(true);
    });

    test("should render the correct `tag` when `tag` prop is passed", () => {
        const rendered: any = shallow(
            <Metatext tag={MetatextTag.p} />
        );
        const paragraph: any = rendered.first().shallow();

        expect(paragraph.instance().props.tag).toEqual(MetatextTag.p);
    });
});
