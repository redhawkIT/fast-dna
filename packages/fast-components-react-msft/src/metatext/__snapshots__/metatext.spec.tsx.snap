// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`metatext snapshots Metatext: 0 1`] = `
<span
  className="typography-0-1-1 typography__7-0-1-8 metatext-0-1-11"
>
  Metatext test string
</span>
`;

exports[`metatext snapshots Metatext: 1 1`] = `
<p
  className="typography-0-1-1 typography__7-0-1-8 metatext-0-1-11"
  data-sketch-symbol="Metatext"
>
  Metatext test string
</p>
`;

exports[`metatext snapshots Metatext: 2 1`] = `
<span
  className="typography-0-1-1 typography__7-0-1-8 metatext-0-1-11"
>
  Metatext test string
</span>
`;
